cloudinit/sources/DataSourceHetzner.py

```diff
@@ -96,9 +106,20 @@ def network_config(self):
         return self._network_config
 
 
-def on_hetzner():
-    return util.read_dmi_data('system-manufacturer') == "Hetzner"
+def get_hcloud_data():
+    vendor_name = util.read_dmi_data('system-manufacturer')
+    if vendor_name != "Hetzner":
+        return (False, None)
```

Comment:
Any  reason for getting rid of the simple 'on_hetzner' function?
It provides an easy way with no side effects to tell if you're on hetzner cloud.

Its not a big deal, just curious why you chose to combine it with this function that also reads the serial.

Reply:
I think you were the one who added it in the first place (after I submitted our DS) to create aa fast exit. Before making this change, I looked at how various other DS handle this  (e.g Googles platform_reports_gce or DO's get_sysinfo). Just wanted to make it sort of consistent with what is already there

Reply:
If we are ever going to go to a "factory" type of solution, then some simple method in each datasoruce that simply returns boolean (running there or not) woudl be useful.

But we're not there yet.